EIPS/eip-5089.md

```diff
@@ -0,0 +1,393 @@
+- slippage: Any difference between advertised redemption value and economic realities of PT redemption, which is not accounted by fees.
+
+### Methods
+
+#### `underlying`
+
+The address of the underlying token used by the Principal Token for accounting, and redeeming.
+
+MUST be an EIP-20 token contract.
+
+MUST _NOT_ revert.
+
+```yaml
+- name: underlying
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: underlyingAddress
+      type: address
+```
+
+#### `maturity`
+
+The unix timestamp (uint256) at or after which Principal Tokens can be redeemed for their underlying deposit.
+
+MUST _NOT_ revert.
+
+```yaml
+- name: maturity
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+    - name: timestamp
+      type: uint256
+```
+
+#### `convertToUnderlying`
+
+The amount of underlying that would be exchanged for the amount of PTs provided, in an ideal scenario where all the conditions are met.
+
+Before maturity, the amount of underlying returned is as if the PTs would be at maturity.
+
+MUST NOT be inclusive of any fees that are charged against redemptions.
+
+MUST NOT show any variations depending on the caller.
+
+MUST NOT reflect slippage or other on-chain conditions, when performing the actual redemption.
+
+MUST NOT revert unless due to integer overflow caused by an unreasonably large input.
+
+MUST round down towards 0.
+
+This calculation MAY NOT reflect the "per-user" price-per-principal-token, and instead should reflect the "average-user's" price-per-principal-token, meaning what the average user should expect to see when exchanging to and from.
+
+```yaml
+- name: convertToUnderlying
+  type: function
+  stateMutability: view
+
+  inputs:
+    - name: principalAmount
+      type: uint256
+
+  outputs:
+    - name: underlyingAmount
+      type: uint256
+```
+
+#### `convertToPrincipal`
+
+The amount of principal tokens that the principal token contract would request for redemption in order to provide the amount of underlying specified, in an ideal scenario where all the conditions are met.
+
+MUST NOT be inclusive of any fees.
+
+MUST NOT show any variations depending on the caller.
+
+MUST NOT reflect slippage or other on-chain conditions, when performing the actual exchange.
+
+MUST NOT revert unless due to integer overflow caused by an unreasonably large input.
+
+MUST round down towards 0.
+
+This calculation MAY NOT reflect the "per-user" price-per-principal-token, and instead should reflect the "average-user's" price-per-principal-token, meaning what the average user should expect to see when redeeming.
+
+```yaml
+- name: convertToPrincipal
+  type: function
+  stateMutability: view
+
+  inputs:
+    - name: underlyingAmount
+      type: uint256
+
+  outputs:
+    - name: principalAmount
+      type: uint256
+```
+
+#### maxRedeem
+
+Maximum amount of principal tokens that can be redeemed from the `owner` balance, through a `redeem` call.
```

Comment:
Should we consider a different term than 'owner' since owner often implies the contract owner when using simple Ownable type contracts.

Reply:
Maybe `holder` would be a better term.

Reply:
`holder` it is, then.